File: 2049-eliminate-maximum-number-of-monsters/2049-eliminate-maximum-number-of-monsters.java
class Solution {
    public int eliminateMaximum(int[] dist, int[] speed) {
        PriorityQueue<Double> pq = new PriorityQueue<>();
        for(int i=0; i<dist.length; i++){
            pq.add(dist[i] * 1.0 / speed[i]);
        }

        double ct = 0;
        int cnt = 0;
        while(pq.size() != 0){
            double rem = pq.remove();
            if(rem <= ct){
                break;
            }
            ct++;
            cnt++;
        }

        return cnt;
    }
}

File: 2049-eliminate-maximum-number-of-monsters/2049-eliminate-maximum-number-of-monsters.cpp
class Solution {
public:
    int eliminateMaximum(vector<int>& dist, vector<int>& speed) {
        priority_queue<double, vector<double>, greater<double>> pq;
        
        for (int i = 0; i < dist.size(); i++) {
            pq.push(static_cast<double>(dist[i]) / speed[i]);
        }

        double ct = 0;
        int cnt = 0;

        while (!pq.empty()) {
            double rem = pq.top();
            pq.pop();
            
            if (rem <= ct) {
                break;
            }
            
            ct++;
            cnt++;
        }

        return cnt;
    }
};